Add projection EPSG support to collection metadata

denarch22 opened the pull request:
## 🛠️ Changes
Introduces a 'projection' field with EPSG code to collection metadata and schema. Automatically extracts EPSG from the first item's properties if not present. Updates validation to enforce presence and type of 'metadata.projection.epsg'.

## 📝 Associated issues
Resolve [LIB-387](https://linear.app/biodev/issue/LIB-387/agregar-informacion-de-proyeccion-a-la-especificacion-de-coleccion)

## ✅ Checks
- [X] Verify that the docs has been updated according to the changes in this PR

denarch22 committed:
Add projection EPSG support to collection metadata

Introduces a 'projection' field with EPSG code to collection metadata and schema. Automatically extracts EPSG from the first item's properties if not present. Updates validation to enforce presence and type of 'metadata.projection.epsg'.

cgalvist reviewed (changes requested):
Hola @denarch22.

Veo que en este PR hay commits de la rama `cgalvis/lib-326`, y en esa rama hay algunos cambios que también afectan la estructura de las colecciones. 

También al ejecutar el comando `python src/format.py`, se cambia el contenido de un archivo.

denarch22 committed:
Update collection metadata specification

Expanded the metadata section to include data_type and projection details. Clarified requirements for properties, values, colors, classes, and class fields based on whether the collection is 'Clasificada' or 'Continua'.

denarch22 committed:
Resolve merge conflict in spec/collection.md

cgalvist reviewed (changes requested):
Hola @denarch22.

Al ejecutar el comando `python src/format.py`, me sigue cambiando el contenido de un archivo.

denarch22 committed:
Refactor EPSG validation for readability

Reformats the conditional checking for 'epsg' in the projection metadata to improve readability by splitting it across multiple lines.

ManuelStardust reviewed (approved):
Probe funcionalidad, las validaciones y la carga del parámetro de poyección espacial estan ok
